pkg/compact/schedule_delete.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package compact
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/thanos-io/thanos/pkg/block"
+	"github.com/thanos-io/thanos/pkg/block/metadata"
+	"github.com/thanos-io/thanos/pkg/objstore"
+	"github.com/thanos-io/thanos/pkg/runutil"
+)
+
+// ScheduleBlockDelete marks the block to be deleted.
+type ScheduleBlockDelete struct {
```

Comment:
Can we try to find a better name for this component? It has nothing to do with scheduling, and also it's an active component so something "BlocksDeleter" or "DelayedBlocksCleaner"?

Reply:
`DelayedBlocksCleaner` looks more apt

Reply:
Lets call it `BlocksCleaner` since `DelayedBlocksCleaner` feels like it cleans delayed blocks

Reply:
`BlocksCleaner` sounds good to me.